napalm_base/mock.py

```
@@ -0,0 +1,200 @@
+            raise TypeError(
+                "{}: expected at most {} arguments, got {}".format(
+                    name, len(parent_method_args.args), len(args) + modifier))
+
+        # Check kwargs
+        unexpected = [x for x in kwargs if x not in parent_method_args.args]
+        if unexpected:
+            raise TypeError("{} got an unexpected keyword argument '{}'".format(name,
+                                                                                unexpected[0]))
+        return mocked_data(path, name, count)
+
+    return _mocked_method
+
+
+def mocked_data(path, name, count):
+    filename = "{}.{}".format(os.path.join(path, name), count)
+    try:
+        with open(filename) as f:
+            result = json.loads(f.read())
+    except IOError:
+        raise NotImplementedError("You can provide mocked data in {}".format(filename))
+
+    if "exception" in result:
+        raise_exception(result)
+    else:
+        return result
+
+
+
+class MockDevice:
+
+    def __init__(self, parent, profile):
+        self.parent = parent
+        self.profile = profile
+
+    def run_commands(self, commands):
+        """Only useful for EOS"""
+        if "eos" in self.profile:
+            return self.parent.cli(commands).values()[0]
+        else:
+            raise AttributeError("MockedDriver instance has not attribute '_rpc'")
+
+
+class MockDriver(NetworkDriver):
+
+    def __init__(self, hostname, username, password, timeout=60, optional_args=None):
+        """
+        Supported optional_args:
+            * path(str) - path to where the mocked files are located
+            * profile(list) - List of profiles to assign
+        """
+        self.hostname = hostname
+        self.username = username
+        self.password = password
+        self.path = optional_args["path"]
+        self.profile = optional_args.get("profile", [])
+
+        self.opened = False
+        self.calls = {}
+        self.device = MockDevice(self, self.profile)
```

Should it raise an exception here if profile is not set (i.e. a blank list)?